pkg/service/group_service.go

```diff
@@ -106,3 +106,260 @@ func (svc *GroupService) DeleteGroup(id string) error {
+//   Required parameters:
+//       id - ID of the group.
+//       srcGroupName - Name of the source group.
+//       srcGroupIp - IP address of the source group.
+//       srcUsername - Username of the source group.
+//       srcPassword - Password of the source group.
+
+//   Optional parameters:
+//       srcPassphrase - Source group encryption passphrase.
+//       skipSecondaryMgmtIp - Skip check for secondary management IP address.
+
+func (svc *GroupService) ValidateMergeGroup(id string, srcGroupName string, srcGroupIp string, srcUsername string, srcPassword string, srcPassphrase *string, skipSecondaryMgmtIp *bool) (*nimbleos.NsGroupMergeReturn, error) {
+
+	if len(id) == 0 || len(srcGroupName) == 0 || len(srcGroupIp) == 0 || len(srcUsername) == 0 || len(srcPassword) == 0 {
+		return nil, fmt.Errorf("error: invalid parameter specified id:%v, srcGroupName:%v, srcGroupIp:%v, srcUsername:%v, srcPassword:%v, srcPassphrase:%v, skipSecondaryMgmtIp:%v ", id, srcGroupName, srcGroupIp, srcUsername, srcPassword, srcPassphrase, skipSecondaryMgmtIp)
+	}
+
+	resp, err := svc.objectSet.ValidateMerge(&id, &srcGroupName, &srcGroupIp, &srcUsername, &srcPassword, srcPassphrase, skipSecondaryMgmtIp)
+	return resp, err
+
+}
+
+// MergeGroup - perform group merge with the specified group.
+//   Required parameters:
+//       id - ID of the group.
+//       srcGroupName - Name of the source group.
+//       srcGroupIp - IP address of the source group.
+//       srcUsername - Username of the source group.
+//       srcPassword - Password of the source group.
+
+//   Optional parameters:
+//       srcPassphrase - Source group encryption passphrase.
+//       force - Ignore warnings and forcibly merge specified group with this group.
+//       skipSecondaryMgmtIp - Skip check for secondary management IP address.
+
+func (svc *GroupService) MergeGroup(id string, srcGroupName string, srcGroupIp string, srcUsername string, srcPassword string, srcPassphrase *string, force *bool, skipSecondaryMgmtIp *bool) (*nimbleos.NsGroupMergeReturn, error) {
+
+	if len(id) == 0 || len(srcGroupName) == 0 || len(srcGroupIp) == 0 || len(srcUsername) == 0 || len(srcPassword) == 0 {
+		return nil, fmt.Errorf("error: invalid parameter specified id:%v, srcGroupName:%v, srcGroupIp:%v, srcUsername:%v, srcPassword:%v, srcPassphrase:%v, force:%v, skipSecondaryMgmtIp:%v ", id, srcGroupName, srcGroupIp, srcUsername, srcPassword, srcPassphrase, force, skipSecondaryMgmtIp)
+	}
+
+	resp, err := svc.objectSet.Merge(&id, &srcGroupName, &srcGroupIp, &srcUsername, &srcPassword, srcPassphrase, force, skipSecondaryMgmtIp)
+	return resp, err
+
+}
+
+// GetEulaGroup - get URL to download EULA contents.
+//   Required parameters:
+//       id - ID of the group.
+
+//   Optional parameters:
+//       locale - Locale of EULA contents. Default is en.
+//       format - Format of EULA contents. Default is HTML.
+//       phase - Phase of EULA contents. Default is setup.
+//       force - Flag to force EULA.
```

Comment:
The default value is specified for the other parameters.  It would be nice to do so for this one and others elsewhere, if feasible.

Reply:
This is straight coming out from sm object XML file, don't think it is feasible.

Reply:
I suspected as much.  This comment should be directed upstream.  :)